# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/VideoEncoder.cpp
# Commit: 45b561797261
# Full Hash: 45b561797261b18d83e66b09c3bf9be9820e4d09
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-03-11 21:13:39
# Description:
#   Bug 1878300 - Only accept vp8 for vp8 codec string. r=chunmin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D200678
# ==============================================================================

diff -r 6901b5e67c87 -r 45b561797261 dom/media/webcodecs/VideoEncoder.cpp
--- a/dom/media/webcodecs/VideoEncoder.cpp	Mon Mar 11 14:01:17 2024 +0000
+++ b/dom/media/webcodecs/VideoEncoder.cpp	Mon Mar 11 14:16:41 2024 +0000
@@ -332,27 +332,6 @@
   return list.forget();
 }
 
-/*
- * The followings are helpers for VideoEncoder methods
- */
-static bool IsEncodeSupportedCodec(const nsAString& aCodec) {
-  LOG("IsEncodeSupported: %s", NS_ConvertUTF16toUTF8(aCodec).get());
-  if (!IsVP9CodecString(aCodec) && !IsVP8CodecString(aCodec) &&
-      !IsH264CodecString(aCodec) && !IsAV1CodecString(aCodec)) {
-    return false;
-  }
-
-  // Gecko allows codec string starts with vp9 or av1 but Webcodecs requires to
-  // starts with av01 and vp09.
-  // https://www.w3.org/TR/webcodecs-codec-registry/#video-codec-registry
-  if (StringBeginsWith(aCodec, u"vp9"_ns) ||
-      StringBeginsWith(aCodec, u"av1"_ns)) {
-    return false;
-  }
-
-  return true;
-}
-
 // https://w3c.github.io/webcodecs/#check-configuration-support
 static bool CanEncode(const RefPtr<VideoEncoderConfigInternal>& aConfig) {
   auto parsedCodecString =
@@ -361,7 +340,7 @@
   if (IsOnAndroid()) {
     return false;
   }
-  if (!IsEncodeSupportedCodec(parsedCodecString)) {
+  if (!IsSupportedVideoCodec(parsedCodecString)) {
     return false;
   }
 